### PR TITLE
[rllib] Use RLlib preprocessors in DQN (fixes PongDeterministic-v4)

### PR DESCRIPTION
**Pull request by ericl**
This fixes the issue reported in https://github.com/ray-project/ray/issues/1113 and https://github.com/ray-project/ray/issues/1028

cc @richardliaw @pcmoritz 

Btw note that to use the same observation dimensions as A3C (42x42), you have to use the following model config:

```
        model:
          grayscale: True
          zero_mean: False
          dim: 42
          conv_filters: [
              [16, [4, 4], 2],
              [32, [4, 4], 2],
              [512, [11, 11], 1],
          ]
```

### CODE REVIEW

File: python/ray/rllib/dqn/models.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def __init__(self, env, config):
             with tf.variable_scope("q_func", reuse=True):
                 q_tp1_using_online_net = _build_q_network(
                     self.obs_tp1, num_actions, config)
-            q_tp1_best_using_online_net = tf.arg_max(q_tp1_using_online_net, 1)
+            q_tp1_best_using_online_net = tf.argmax(q_tp1_using_online_net, 1)
```
Comment (ericl) [inline python/ray/rllib/dqn/models.py]: Apparently the old syntax is deprecated

File: python/ray/rllib/dqn/common/atari_wrappers_deprecated.py [hunk 2]
```diff
@@ -211,41 +213,23 @@ def _get_ob(self):
...
 
-class A2cProcessFrame(gym.Wrapper):
-    def __init__(self, env):
-        gym.Wrapper.__init__(self, env)
-        self.observation_space = spaces.Box(low=0, high=255, shape=(80, 80, 1))
+    if is_atari:
+        env = EpisodicLifeEnv(env)
+        env = NoopResetEnv(env, noop_max=30)
+        if 'NoFrameskip' in env.spec.id:
+            env = MaxAndSkipEnv(env, skip=4)
```
Comment (ericl) [inline python/ray/rllib/dqn/common/atari_wrappers_deprecated.py]: We should consider also using these wrappers with other envs I think.

File: python/ray/rllib/dqn/dqn.py [hunk 3]
```diff
@@ -12,8 +12,7 @@
 import ray
 from ray.rllib.common import Agent, TrainingResult
 from ray.rllib.dqn import logger, models
-from ray.rllib.dqn.common.atari_wrappers_deprecated \
-    import wrap_dqn, ScaledFloatFrame
+from ray.rllib.dqn.common.atari_wrappers_deprecated import wrap_dqn
```
Comment (richardliaw) [inline python/ray/rllib/dqn/dqn.py]: I understand the "deprecated" part is vestigal from OpenAI's codebase, but should we keep it?

File: python/ray/rllib/dqn/common/atari_wrappers_deprecated.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/rllib/dqn/common/atari_wrappers_deprecated.py]: Are any of these Pong specific?

Comment (ericl) [reply python/ray/rllib/dqn/common/atari_wrappers_deprecated.py]: I think they're general for atari.

File: python/ray/rllib/dqn/dqn.py (hunk 3 above)
Comment (ericl) [reply python/ray/rllib/dqn/dqn.py]: Renamed

File: python/ray/rllib/models/catalog.py [hunk 4]
```diff
@@ -34,6 +35,9 @@ class ModelCatalog(object):
         action_op = dist.sample()
     """
 
+    ATARI_OBS_SHAPE = (210, 160, 3)
```
Comment (richardliaw) [inline python/ray/rllib/models/catalog.py]: A little odd that these dimensions are in the Model Catalog, but it's fine for now.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Will merge after last build.

**Pull request merged**